File: pinpointB/src/main/java/com/nxquar/pinpoint/DTO/AdminRequest.java
package com.nxquar.pinpoint.DTO;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.nxquar.pinpoint.Model.Address;
import com.nxquar.pinpoint.Model.Users.Institute;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class AdminRequest {
    private String name;
    private String email;
    private String phone;
    private String password;  // Optional field for updates
    private Address address;
    private UUID instituteId;
}

File: pinpoint/lib/model/admin/admin_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pinpoint/model/address/address.dart';

part 'admin_request.g.dart';

@JsonSerializable(includeIfNull: false)
class AdminRequest {
  final String? name;
  final String? email;
  final String? phone;
  final String? password;
  final Address? address;
  final String? instituteId;

  AdminRequest({
    this.name,
    this.email,
    this.phone,
    this.password,
    this.address,
    this.instituteId,
  });

  factory AdminRequest.fromJson(Map<String, dynamic> json) =>
      _$AdminRequestFromJson(json);

  Map<String, dynamic> toJson() => _$AdminRequestToJson(this);
}
